### DISCUSSION TITLE
LED-Light with GPIO13? El. Powermeter least digit resolution?

### DISCUSSION THREAD

**Discussion by Franz4596** [jomjol/AI-on-the-edge-device#3015] [Q&A]
I kindly ask for help of the community: 

1. There are to much reflections due to the Meter Glaswindow with the onboard LED light. So I do a external lightning with two usual white LEDs positioned near the Meter surface (see photo). But the LEDs are connected to 5V (270Ohm) all the time. Is it possible to switch the LEDs on/off with eg GPIO13? How is Configuration? Max of the GPIO current ?  

2. I measure an Electric  Powermeter, all 7 digits are recognized well, including least the fraction of the least digit (eg. 2.5). But the least digit was stored as a full number (2) not with the full resolution (2.5). Is ist possible do save the fraction too?
![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/62063765/e42b6480-a9a0-4ff2-80c1-66daee61c308)

**Comment by SybexX** [comment]
to 1. https://esp32.com/viewtopic.php?t=2384
https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#section-gpio

to 2. https://jomjol.github.io/AI-on-the-edge-device-docs/Correction%20Algorithm/#extendedresolution

**Reply by Franz4596** [reply]
thanks for the quick answer and helpful links !
to 1. -> "the maximum one can safely draw from a GPIO pin is 12mA", means one LED with 100 Ohm will work (3,3-2)V/12mA 
        -> I did read the "#section-gpio", and tested "GPIO12 div. configuration", but a usual LED do not work. 
            So is seams I have to order a WS2812 which will be controlled by GOIO12 (or alternativly change Program to output) 

to 2.  at last I used the predefined model "dig-cont_6020_s3.tflite", which works good (except last digit resolution)
Now I select "dig-class100-168_s3.tflite":
![Bild](https://github.com/jomjol/AI-on-the-edge-device/assets/62063765/49c639ea-9a29-408c-8eaf-c4880e8acb8f)
but the value of class100-model is same 79039.6 and not 79039.6**6** ! 
shall I config an other model or any thing else?

**Reply by Franz4596** [reply]
Additional Notes:
🥇 to 2.: The solution is: setting "**Extended Resolution = TRUE**" 👍 :
![Bild1](https://github.com/jomjol/AI-on-the-edge-device/assets/62063765/fa7a878e-7ca0-441f-8066-e466b0e60ef8)

to 1.: A usual LED do not work, GPIO12 WS281X-configured, flashs short only, gives small pulses, therfore an additional hardware (eg FlipFlop) is needed, or a change of firmeware.

**Reply by SybexX** [reply]
WS281X is not controlled by a classic pwm, a protocol is issued to control the WS281X
If you want you can test this customized firmware:
[AI-on-the-edge-device__update__cam_settings_(1e2d18d).zip](https://github.com/jomjol/AI-on-the-edge-device/files/14827217/AI-on-the-edge-device__update__cam_settings_.1e2d18d.zip)

![Unbenannt](https://github.com/jomjol/AI-on-the-edge-device/assets/587201/3e69e8ce-8ddc-494a-808c-738f96f9ec00)
